gatewayconfig/gpio/gpio_button.py

```diff
@@ -0,0 +1,78 @@
+import time
+import threading
+
+from gpio4 import SysfsGPIO
+
+from hm_pyhelper.logger import get_logger
+
+
+LOGGER = get_logger(__name__)
+
+
+# How often to check for button presses
+MONITOR_CHECK_INTERVAL_SECONDS = 0.05
+
+
+class GpioButton(threading.Thread):
+    def __init__(self, pin_number, hold_seconds):
+        super(GpioButton, self).__init__()
+        LOGGER.info("Initializing GPIO button on pin %s" % pin_number)
+        self.daemon = True
+        self.cancelled = False
+        self.gpio = SysfsGPIO(pin_number)
+        self.hold_seconds = hold_seconds
+        self.when_held = None
+        self.reset_pressed_state()
+        self.gpio.export = True
+        self.gpio.direction = 'in'
+
+    def run(self):
+        while not self.cancelled:
+            if not self.is_pressed():
+                self.reset_pressed_state()
+            else:
+                self.process_press()
+
+            time.sleep(MONITOR_CHECK_INTERVAL_SECONDS)
+
+    def process_press(self):
+        """
+        If this is the first time this press has been detected,
+        record the time for future reference. If the press has
+        already been detected, attempt to trigger #when_held.
+        """
+
+        if self.last_pressed_at is None:
+            LOGGER.debug("Button pressed for first time")
+            self.last_pressed_at = time.time()
+        else:
+            self.trigger_when_held_after_hold_seconds()
+
+    def trigger_when_held_after_hold_seconds(self):
```

Comment:
can we move any duplicated stuff in here / mraa_button.py into a like button_common.py file or similar and just pull in those functions rather than duplicating?

Reply:
Agreed. Done!